tests/suite/nginxgateway_test.go

```diff
@@ -0,0 +1,250 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	core "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	ngfAPI "github.com/nginxinc/nginx-gateway-fabric/apis/v1alpha1"
+	"github.com/nginxinc/nginx-gateway-fabric/tests/framework"
+)
+
+var _ = Describe("NginxGateway", Ordered, Label("functional", "nginxGateway"), func() {
+	var (
+		ngfPodName string
+
+		namespace          = "nginx-gateway"
+		nginxGatewayNsname = types.NamespacedName{Name: releaseName + "-config", Namespace: namespace}
+
+		files = []string{
+			"nginxgateway/nginx-gateway.yaml",
+		}
+	)
+
+	getNginxGateway := func(nsname types.NamespacedName) (ngfAPI.NginxGateway, error) {
+		ctx, cancel := context.WithTimeout(context.Background(), timeoutConfig.GetTimeout)
+		defer cancel()
+
+		var nginxGateway ngfAPI.NginxGateway
+
+		if err := k8sClient.Get(ctx, nsname, &nginxGateway); err != nil {
+			return nginxGateway, errors.New("failed to get nginxGateway")
+		}
+
+		return nginxGateway, nil
+	}
+
+	verifyNginxGateway := func(nsname types.NamespacedName, expObservedGen int64) error {
+		nginxGateway, err := getNginxGateway(nsname)
+		if err != nil {
+			return err
+		}
+
+		if nginxGateway.Status.Conditions == nil {
+			return errors.New("nginxGateway is has no conditions")
+		}
+
+		if len(nginxGateway.Status.Conditions) != 1 {
+			return fmt.Errorf("expected nginxGateway to have only one condition, instead has %d conditions",
```

Comment:
nit: when breaking lines up, each arg should be on its own line:

```
return fmt.Errorf(
    "expected nginxGateway to have only one condition, instead has %d conditions",
    len(nginxGateway.Status.Conditions), 
 )
```

same goes for the rest of the function